drivers/serial/uart_nrfx_uarte.c

```diff
@@ -170,32 +204,46 @@ static inline NRF_UARTE_Type *get_uarte_instance(const struct device *dev)
 static void uarte_nrfx_isr_int(void *arg)
 {
 	const struct device *dev = arg;
-	struct uarte_nrfx_data *data = get_dev_data(dev);
 	NRF_UARTE_Type *uarte = get_uarte_instance(dev);
 
-	if (data->int_driven->disable_tx_irq &&
-	    nrf_uarte_event_check(uarte, NRF_UARTE_EVENT_ENDTX)) {
-		nrf_uarte_int_disable(uarte, NRF_UARTE_INT_ENDTX_MASK);
-
-		/* If there is nothing to send, driver will save an energy
-		 * when TX is stopped.
-		 */
-		nrf_uarte_task_trigger(uarte, NRF_UARTE_TASK_STOPTX);
+	/* If interrupt driven and asynchronous APIs are disabled then UART
+	 * interrupt is still called to stop TX. Unless it is done using PPI.
+	 */
+	if (nrf_uarte_int_enable_check(uarte, NRF_UARTE_INT_ENDTX_MASK) &&
+		nrf_uarte_event_check(uarte, NRF_UARTE_EVENT_ENDTX)) {
+		endtx_isr(dev);
+		return;
```

Comment:
This `return` should be removed. The handler should exit here only when interrupt driven API is not used for the instance, but this case is handled in [line 221](https://github.com/zephyrproject-rtos/zephyr/pull/28961/commits/834695f119ecdf82f44f29831b8bb13c396e0629#diff-0de7d1227701470077e7117a57033444R221).

Reply:
done